test/e2e/operator_test.go

```diff
@@ -3579,6 +3579,47 @@ func waitForDeploymentEnvVar(t *testing.T, cl client.Client, deployment *appsv1.
 	return err
 }
 
+// waitForDeploymentCompleteWithCleanup waits for a deployment to complete its rollout, then waits for the old
+// generation's pods to finish terminating.
+func waitForDeploymentCompleteWithOldPodTermination(t *testing.T, cl client.Client, deploymentName types.NamespacedName, timeout time.Duration) error {
+	t.Helper()
+	deployment := &appsv1.Deployment{}
+	if err := cl.Get(context.TODO(), deploymentName, deployment); err != nil {
+		return fmt.Errorf("failed to get deployment %s: %w", deploymentName.Name, err)
+	}
+
+	if err := waitForDeploymentComplete(t, cl, deployment, timeout); err != nil {
+		return fmt.Errorf("timed out waiting for deployment %s to complete rollout: %w", deploymentName.Name, err)
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return fmt.Errorf("deployment %s has invalid spec.selector: %w", deploymentName.Name, err)
+	}
+
+	expectedReplicas := 1
+	if deployment.Spec.Replicas != nil && int(*deployment.Spec.Replicas) != 0 {
```

Comment:
The `int(*deployment.Spec.Replicas) != 0` seems a little funny to me. `replicas: 0` is indeed an allowed value in an deployment and if you did have a deployment with it, this function would fail since it overrides it to 1.

I know it's a highly unlikely use-case. Did you have another reason for ignoring `replicas: 0` and overriding it to 1?

Reply:
Fixed in https://github.com/openshift/cluster-ingress-operator/pull/904/commits/39c6aef16c999bc4e05e948238454439e8c21b3d.